docs/tools/plist.md

```diff
@@ -0,0 +1,301 @@
+# Static Analysis PLIST Format Specification
+
+Plist files are used in CodeChecker to store and parse diagnostic 
+information from static analysis tools.
+This documentation explains the structure of plist files generated by 
+CodeChecker, illustrating how they are represented in the code and what 
+they look like on the CodeChecker UI.
+
+## Table of Contents
+* [Plist files in CodeChecker](#plist-files-in-codechecker)
+* [Structure of a .plist File](#structure-of-a-plist-file)
+* [Diagnostic Entry Examples](#diagnostic-entry-examples)
+  * [1. `bug_path_event` — Bug Path Events](#1-bug_path_event--bug-path-events)
+  * [2. `notes` — Notes](#2-notes--notes)
+  * [3. `macro_expansions` — Macro Expansions](#3-macro_expansions--macro-expansions)
+  * [4. `report-annotation` — Report Annotations](#4-report-annotation--report-annotations)
+* [License](#license)
+
+
+## Plist files in CodeChecker
+
```

Comment:
Maybe you can start with this: Plist is the default static and dynamic analysis inter-change file format of CodeChecker. They are created as a results of the analysis: `CodeChecker analyze -o report_directory" invocation in the report directory. The proprietary report format of each invoked analyzer tool is converted into `plist` format, by the report-converter tools. Then these plist reports are consumed by the `CodeCheckecker "store/diff/parse` subcommands.